test/cql-pytest/test_secondary_index.py

```diff
@@ -100,3 +100,36 @@ def index_used(query, index_name):
         # Local indexes are still preferred over global ones, if they can be used
         index_used(f"SELECT * FROM {table} WHERE p = 1 and v1 = 1 and v3 = 2 and v2 = 2 allow filtering", "my_v2_idx")
         index_used(f"SELECT * FROM {table} WHERE p = 1 and v2 = 1 and v1 = 2 allow filtering", "my_v2_idx")
+
+# Indexes can be created without an explicit name, in which case a default name is chosen.
+# However, due to #8620 it was possible to break the index creation mechanism by creating
+# a properly named regular table, which conflicts with the generated index name.
+# A similar issue still exists for named indexes, so both cases are checked in this scenario.
+def test_create_index_when_its_name_is_taken(scylla_only, cql, test_keyspace):
+    schema = 'p int primary key, v int'
+    with new_test_table(cql, test_keyspace, schema) as table:
+        # In case of default names, a proper name can always eventually be picked
+        index_created = False
+        try:
+            cql.execute(f"CREATE TABLE {table}_v_idx_index (i_do_not_exist_in_the_base_table int primary key)")
+            # Creating an index should succeed, even though its default name is taken
+            # by the table above
+            cql.execute(f"CREATE INDEX ON {table}(v)")
+            index_created = True
+            cql.execute(f"DROP INDEX {table}_v_idx_1")
+        finally:
+            cql.execute(f"DROP TABLE {table}_v_idx_index")
+        assert index_created
```

Comment:
I don't think you need this assert or the "index_created" variable. The only way for index_created to remain false is if the above code threw an exception and didn't run all the way. But if it did, the whole function will be exited with this exception! (the finally is guaranteed to run, but the function doesn't continue after it).